src/qt/sendcoinsdialog.cpp

```diff
@@ -502,19 +502,34 @@ void SendCoinsDialog::sendButtonClicked([[maybe_unused]] bool checked)
     }
 
     bool send_failure = false;
-    if (retval == QMessageBox::Save) {
+    bool broadcast = true;
```

Comment:
33d165d802846943c0ba70be4e643e3b50c6634e

```diff
diff --git a/src/qt/sendcoinsdialog.cpp b/src/qt/sendcoinsdialog.cpp
index c420aa3581..87a0e4a098 100644
--- a/src/qt/sendcoinsdialog.cpp
+++ b/src/qt/sendcoinsdialog.cpp
@@ -502,7 +502,6 @@ void SendCoinsDialog::sendButtonClicked([[maybe_unused]] bool checked)
     }

     bool send_failure = false;
-    bool broadcast = true;
     if (retval == QMessageBox::Save) { // Create Unsigned clicked
         CMutableTransaction mtx = CMutableTransaction{*(m_current_transaction->getWtx())};
         PartiallySignedTransaction psbtx(mtx);
@@ -510,11 +509,13 @@ void SendCoinsDialog::sendButtonClicked([[maybe_unused]] bool checked)
         presentPSBT(psbtx);
     } else { // Send clicked
         assert(!model->wallet().privateKeysDisabled() || model->wallet().hasExternalSigner());
+        bool broadcast = true;
         if (model->wallet().hasExternalSigner()) {
             CMutableTransaction mtx = CMutableTransaction{*(m_current_transaction->getWtx())};
             PartiallySignedTransaction psbtx(mtx);
             bool complete = false;
             send_failure = !signWithExternalSigner(psbtx, mtx, complete);
+            broadcast = complete;
             // A transaction signed with an external signer is not always complete,
             // e.g. in a multisig wallet.
             if (complete) {
@@ -523,7 +524,6 @@ void SendCoinsDialog::sendButtonClicked([[maybe_unused]] bool checked)
                 m_current_transaction->setWtx(tx);
             } else if (!send_failure) {
                 presentPSBT(psbtx);
-                broadcast = false;
             }
         }
```

Reply:
Oops...

Reply:
If the user rejects the transaction then `complete = false` and it will present the PSBT. We can't tell the difference between rejection and failure, so that's difficult to avoid. I don't see how this patch would change that.

Reply:
If user rejects then `complete = false` and `send_failure = true`, so it doesn't `presentPSBT()` and because 
`broadcast = true` it calls `model->sendCoins()`

Reply:
I see. I restructured this code and fix a bit...

Reply:
Thanks, LGTM, will re-test.